src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs

```diff
@@ -499,7 +469,7 @@ private void WriteTaggableCommonMethodBranch(MgmtRestOperation operation, IEnume
 
             var newInstanceExpression = _resource.NewInstanceExpression(new[]
             {
-                new ParameterInvocation(_resource.OptionsParameter, w => w.Append($"this")),
+                new ParameterInvocation(_resource.ResourceParameter, w => w.Append($"ArmClient")),
```

Comment:
It this the same `ArmClient` as in the `ArmClientReference`?

Reply:
ResourceCollectionWriter has similar change but doesn't override `ArmClientReference`: https://github.com/Azure/autorest.csharp/pull/1896/files#diff-bd1c36627623a9f4e5439e3a5b022fe5208c14f94874ab7ef59b0049afc60662R285